src/services/infra/InfraService.ts

```diff
@@ -91,4 +105,174 @@ export default class InfraService {
       throw err
     }
   }
+
+  parseDNSRecords = (record?: string) => {
```

Comment:
hmm I have a soft preference for this. Conversely, having a stronger API contract means that every time I call this method, I am going to have check the variable to see if it is undefined. Ie an extra `if` every time I call this method rather than just 1

Reply:
your point is true! but it's important to note here that we don't do anything meaningful with the `undefined` type - we simply return as-is and right after, we do a 
```ts
const record = parseDNSRecords(s)
if (!record) { 
 // log some error
} 
```
if we do want to preserve this API contract, we could instead ask that this return a meaningful error which allows us to distinguish between `record` being absent or `record` being malformed

Reply:
04d1450